src/Offer/ReadModel/JSONLD/OfferLDProjector.php

```diff
@@ -422,7 +422,7 @@ protected function applyImageRemoved(AbstractImageRemoved $imageRemoved): ?JsonD
         );
 
         // Unset the main image if it matches the removed image
-        if (isset($offerLd->image) && strpos($offerLd->{'image'}, $imageId)) {
+        if (isset($offerLd->image)) {
```

Comment:
By doing the `unset` of the main image always I think that the main image will be reset to the first image. Even when the main image was not removed.

I think this can be tested by having three images, the main image being the third image. Then a removal of the second image will result in the first image becoming the main image. But this should stay the last image.

Reply:
This is now fixed (for both udb2 & udb3 images) by checking the url instead of the id.